src/battle.ts

```diff
@@ -2381,8 +2381,10 @@ export class Battle {
 			let species = Dex.species.get(args[2]);
 			let fromeffect = Dex.getEffect(kwArgs.from);
 			let isCustomAnim = species.name.startsWith('Wishiwashi');
-			poke.removeVolatile('typeadd' as ID);
-			poke.removeVolatile('typechange' as ID);
+			if (!poke.getSpeciesForme().endsWith('-Gmax') && !species.name.endsWith('-Gmax')) {
+				poke.removeVolatile('typeadd' as ID);
+				poke.removeVolatile('typechange' as ID);
+			}
 			if (this.gen >= 7) poke.removeVolatile('autotomize' as ID);
```

Comment:
should I move the autotomize thing into the if statement as well (and also change it to check for gen 6 cause that's when forme changes started to get rid of autotomize according to bulbapedia)

Reply:
Yep, they're not forme changes so weight's not reset either. That generation check isn't even being done server-side as far as I can tell. It is Gen 6 onward.

Reply:
> That generation check isn't even being done server-side as far as I can tell.

Server side, Gen 6+ autotomize directly modifies the Pokémon's weight, which means that a forme change will get rid of the effect, while Gen 5 adds a volatile (that isn't copied by Baton Pass) which reduces the weight.